management/hook_secrets.go

```diff
@@ -0,0 +1,40 @@
+package management
+
+type HookSecrets = map[string]string
+
+type HookSecretsManager struct {
+	*Management
+}
+
+func newHookSecretsManager(m *Management) *HookSecretsManager {
+	return &HookSecretsManager{m}
+}
+
+// Upserts hook secrets
+//
+// See: https://auth0.com/docs/api/management/v2#!/Hooks/post_secrets
+func (m *HookSecretsManager) Upsert(hookId string, r *HookSecrets) (err error) {
+	return m.post(m.hookPath(hookId), r)
+}
```

Comment:
[`POST /api/v2/hooks/{id}/secrets`](https://auth0.com/docs/api/management/v2#!/Hooks/post_secrets) doesn't have upsert semantics. Posting a map that contains an existing field will result in a conflict.

```
{
  "statusCode": 409,
  "error": "Conflict",
  "message": "Error: Secret exists",
  "errorCode": "SecretExists"
}
```

Reply:
OMG It's even worse than that. The PATCH doesn't work if you specify a new secret field.  I found the only way to add a new secret to a hook that already has a secret is to completely DELETE all secrets for that hook then "POST" with all keys and values (even though you can't retrieve the values via the API).  I've got a ticket open with Auth0 support about this.